Source/NSMenu.m

```diff
@@ -1860,6 +1869,38 @@ - (void) _showOnActivateApp: (NSNotification*)notification
   }
 }
 
+- (void) windowDidChangeScreen: (NSNotification*)notification
+{
+  NSWindow *window = [notification object];
+  NSRect   frame;
+  NSRect   oldScreenFrame;
+  NSRect   newScreenFrame;
+  NSPoint  offset;
+
+  if (window == _aWindow
+      || [window isKeyWindow] == NO
+      || [_aWindow screen] == [window screen]
+      || [_aWindow isVisible] == NO)
+    {
+      return;
+    }
+
+  oldScreenFrame = [[_aWindow screen] frame];
+  newScreenFrame = [[window screen] frame];
+  frame = [_aWindow frame];
+  
+  // Keep left offset fixed
+  offset.x = frame.origin.x - oldScreenFrame.origin.x;
+  frame.origin.x = newScreenFrame.origin.x + offset.x;
```

Comment:
I think it would be more clear to write this as

frame.origin.x += newScreenFrame.origin.x - oldScreenFrame.origin.x;

The y direction is not as simple.

Reply:
Done.